docs/source/guides/monitoring.rst

```diff
@@ -1,12 +1,585 @@
+==========================
 Monitoring with Prometheus
 ==========================
 
+Monitoring stacks usually consist of a metrics collector, a time-series database to store metrics
+and a visualization layer. A popular stack is Prometheus_ with Grafana_ used as the visualization layer
+to create rich dashboards. An architecture of Prometheus and its ecosystem is shown below:
+
+.. image:: ../_static/img/prom-architecture.png
+
+BentoML API server comes with Prometheus support out of the box. When launching an API model server with BentoML,
+whether it is running dev server locally or deployed with docker in the cloud, a ``/metrics`` endpoint will always
+be available, which includes the essential metrics for model serving and the ability to create
+and customize new metrics base on needs. This guide will introduce how you use Prometheus and Grafana to monitor
+your BentoService.
+
+Preface
+-------
+
+.. seealso::
+     Prometheus_ and Grafana_ docs for more in depth topics.
+
+.. note::
+    This guide requires users to have a basic understanding of Prometheus' concept as well as its metrics type.
+    Please refers to `Concepts <https://prometheus.io/docs/concepts/data_model/>`_ for more information.
+
+.. note::
+    Refers to `PromQL basics <https://prometheus.io/docs/prometheus/latest/querying/basics/>`_ for Prometheus query language.
+
+.. note::
+    Please refers to Prometheus' best practices for `consoles and dashboards <https://prometheus.io/docs/practices/consoles/>`_
+    as well as `histogram and summaries <https://prometheus.io/docs/practices/histograms/>`_.
+
+Local Deployment
+----------------
+
+This section will walk you through how to setup the stack locally, with the optional guide on using ``docker-compose`` for easy deployment of the stack.
+
+
+Setting up Prometheus
+^^^^^^^^^^^^^^^^^^^^^
+
+It is recommended to run Prometheus with Docker . Please make sure that you have
+`Docker <https://docs.docker.com/engine/install/>`_ installed on your system.
+
+Users can take advantage of having a ``prometheus.yml`` for configuration.
+Example of for monitoring BentoService is shown below:
+
+.. code-block:: yaml
+
+    # prometheus.yml
+
+    global:
+      scrape_interval:     15s
+      evaluation_interval: 30s
+      # scrape_timeout is set to the global default (10s).
+
+    scrape_configs:
+    - job_name: prometheus
+
+      honor_labels: true
+      static_configs:
+      - targets:
+        - localhost:9090  # metrics from prom/prometheus
+        - localhost:5000  # BentoService metrics
+        - localhost:3000  # YataiService metrics
```

Comment:
Usually, the Yatai and BentoService are deployed separately and their monitoring needs are different as well, maybe let's focus on monitoring BentoService for this guide? 

It actually will be more useful to include monitoring for multiple BentoServices with different names

Reply:
Will do.